lib/releaselib.sh

```diff
@@ -911,17 +912,17 @@ release::docker::release () {
     "hyperkube"
   )
 
-  if [[ -n $G_AUTH_USER && $registry == "gcr.io/google_containers" ]];then
-    logrun $GCLOUD config set account $G_AUTH_USER
+  if [[ $registry == "$GCRIO_PATH_PROD" ]];then
+    push_registry="$GCRIO_PATH_PROD_PUSH"
+
+    [[ -n $G_AUTH_USER ]] && logrun $GCLOUD config set account $G_AUTH_USER
   fi
 
+  docker_push_cmd=("$GCLOUD" "docker" "-s" "$push_registry" "--")
```

Comment:
I think you might need to munge `$push_registry` for non-prod cases, too; for example, `push-build.sh` is called from test-infra and passes `gcr.io/kubernetes-ci-images` as the registry, and that's not a valid server name. (`gcr.io` is, though.)

Reply:
actually I don't think you need to pass the `-s` arg at all anymore with newer versions of `gcloud`.

(a separate issue is that the `gcloud docker` wrapper is going away, to be replaced by `gcloud beta auth configure-docker` followed by regular `docker push`.)

Reply:
Thanks for the heads up.  Have you dealt with this yet in test-infra?  Just attempting to "`gcloud beta auth configure-docker`" puts me into an interactive 'command group' installation situation.  
Thinking it might be a good idea here to write something that automatically detects and switches gears as needed, but there are many layers.  Anything I can borrow here or I can build it out.